src/node_contextify.cc

```diff
@@ -1582,6 +1589,15 @@ Local<Object> ContextifyContext::CompileFunctionAndCacheResult(
   Local<Object> result = Object::New(isolate);
   if (result->Set(parsing_context, env->function_string(), fn).IsNothing())
     return Object::New(env->isolate());
+
+  // ScriptOrigin::ResourceName() returns SourceURL magic comment content if
+  // present.
+  if (result
+          ->Set(parsing_context,
+                env->source_url_string(),
+                fn->GetScriptOrigin().ResourceName())
+          .IsNothing())
+    return Object::New(env->isolate());
```

Comment:
hmm... if `IsNothing()` is true that would suggest that an error is scheduled. We probably shouldn't return a new object in that case and should instead allow the error to propagate?

Also, this likely needs surrounding `{ }` to improve readability.

Reply:
This will be better changed together with the return type corrected: https://github.com/nodejs/node/pull/58434.